src/quacc/recipes/espresso/bands.py

```diff
@@ -34,7 +34,7 @@ class BandsSchema(TypedDict, total=False):
 @job
 def bands_pw_job(
     atoms: Atoms,
-    copy_files: SourceDirectory | dict[SourceDirectory, Filenames],
+    copy_files: SourceDirectorySchema,
```

Comment:
Personally, I'm not a fan of this type hinting approach throughout the module. It is not immediately obvious to the user what this is. Previously, we had

```python
SourceDirectory | dict[SourceDirectory, Filenames]
```

The user can likely know at a glance what a `SourceDirectory` might be or how the dictionary would be formatted, without having to dig through the codebase. However, a `SourceDirectorySchema` is quite niche. Schema also implies that it is the output of a calculation, which doesn't seem to be the case here.

Can you clarify your goal here? I can then provide a more appropriate recommendation.

Reply:
@tomdemeyere: I think this was a major source of my confusion for several reasons.

You have defined this `SourceDirectorySchema` as

```python
SourceDirectory | dict[str, SourceDirectory] | list[SourceDirectory]
```

However, this doesn't seem right now that I read this closer. The type hint seems like it should _actually_ be

```python
SourceDirectory | list[SourceDirectory] | dict[SourceDirectory, Filenames]
```

Is that correct? Note how there is no `dict[str, SourceDirectory]`. I do not know what that would be.

Assuming I am correct here, please use the `SourceDirectory | list[SourceDirectory] | dict[SourceDirectory, Filenames]` type hint throughout. Collapsing it all to one type hint anyway is simply more confusing to users.